json-jackson-v2/src/main/java/tech/ydb/yoj/repository/db/json/JacksonJsonConverter.java

```diff
@@ -0,0 +1,98 @@
+package tech.ydb.yoj.repository.db.json;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.base.Suppliers;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import tech.ydb.yoj.repository.db.common.JsonConverter;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.function.Supplier;
+
+/**
+ * {@link JsonConverter YOJ JSON Converter} implementation using Jackson as the underlying JSON library.
+ * Use it to support JSON-valued fields ({@link tech.ydb.yoj.databind.schema.Column @Column(flatten=false)} composite
+ * objects and dynamic fields with type of interface/abstract class, e.g. {@link java.util.List}):
+ * <blockquote>
+ * <pre>
+ * CommonConverters.defineJsonConverter(new JacksonJsonConverter());
+ * </pre>
+ * </blockquote>
+ * <p>
+ * To customize the {@link ObjectMapper} being created, create a subclass of {@code JacksonJsonConverter} and override
+ * its {@link #createObjectMapper()} method.
+ */
+public class JacksonJsonConverter implements JsonConverter {
+    private final Supplier<ObjectMapper> mapper = Suppliers.memoize(this::createObjectMapper);
+
+    @Override
+    @SneakyThrows
+    public final String toJson(@NonNull Type type, @Nullable Object o) {
+        return mapper().writerFor(mapper().getTypeFactory().constructType(type)).writeValueAsString(o);
+    }
+
+    @Override
+    @SneakyThrows
+    public final Object fromJson(@NonNull Type type, @NonNull String content) {
+        return mapper().readerFor(mapper().getTypeFactory().constructType(type)).readValue(content);
+    }
+
+    @Override
+    @SneakyThrows
+    public final Object fromObject(@NonNull Type type, @Nullable Object content) {
+        JavaType jacksonType = mapper().getTypeFactory().constructType(type);
+        return content != null
+                ? mapper().convertValue(content, jacksonType)
+                : (jacksonType.isCollectionLikeType() ? List.of() : Map.of());
+    }
+
+    public String toString() {
+        return getClass().getSimpleName();
+    }
+    
+    protected final ObjectMapper mapper() {
+        return mapper.get();
+    }
+
+    protected ObjectMapper createObjectMapper() {
```

Comment:
nit: To prohibit the use of inheritance, I think it is better to make the class private, remove protected methods and make extension through the JsonConverter interface, not through inheritance. I could look better if we make this function like static public, no-args constructor like { this(createObjectMapper()); } and main constructor like JacksonJsonConverter(ObjectMapper mapper) { this.mapper = mapper)

In this case you still can make your own realization on delegates.

Reply:
Yep! That's primarily to not call protected `createObjectMapper()` method in the class's constructor, otherwise the subclass will not be initialized when `createObjectMapper()` is called, leaving some fields with their default values.

e.g.
```java
public class XConverter extends JacksonJsonConverter {
   private final int x;

   public XConverter(int x) {
      super();
      this.x = x;
   }

   protected ObjectMapper createObjectMapper() {
      // x will be 0 here because protected method is called by superclass
      System.out.println(x);
      return super.createObjectMapper();
   }
```

Also to not construct the `ObjectMapper` unless its needed, but this is a secondary concern here.

Alternatively, users can pass `Supplier<ObjectMapper>` to constructor of `final class JacksonJsonConverter` instead of overriding `createObjectMapper()` in a subclass, but then they wouldn't be able to add any additional attributes to the class.

Reply:
Fixed